constantfold.ml

```diff
@@ -28,6 +27,17 @@ let const_prop prog =
   let convert x l instr =
     let replace = Replace.var_in_exp x l in
     match instr with
+    | Call (y, f, es) ->
+      assert (x <> y);
+      let replace_arg = function
```

Comment:
Should we move this to `replace.ml` (since `rename.ml` has a similarly short function for renaming in an arg)?